test/unit/mocha.spec.js

```diff
@@ -446,6 +446,36 @@ describe('Mocha', function() {
       var mocha = new Mocha(opts);
       expect(mocha.reporter(), 'to be', mocha);
     });
+
+    it('should keep reporterOption on options', function() {
+      var mocha = new Mocha({
+        reporter: 'spec',
+        reporterOption: {
+          foo: 'bar'
+        }
+      });
+      expect(mocha.options.reporterOption, 'to have property', 'foo', 'bar');
+    });
```

Comment:
pls merge the first two tests into one and add:
`expect(mocha.options.reporterOptions, 'to have property', 'foo', 'bar');`